src/cc_ring_array.c

```diff
@@ -162,8 +170,9 @@ ring_array_create(size_t elem_size, uint32_t cap)
 }
 
 void
-ring_array_destroy(struct ring_array *arr)
+ring_array_destroy(struct ring_array **arr)
 {
-    log_verb("destroying ring array %p and freeing memory");
-    cc_free(arr);
+    log_verb("destroying ring array %p and freeing memory", *arr);
```

Comment:
should probably check for NULL pointer here

```
if ((arr == NULL) || (*arr == NULL)) {
    log_error("ring_array_destroy passed NULL pointer");
    return;
```

(or if we want to be especially bold make it an assertion)

Reply:
good call, thanks!